src/components/Shared/ReceiveModal/ReceiveModal.tsx

```diff
@@ -125,46 +148,56 @@ const ReceiveModal: FC<ReceiveModalProps> = (props) => {
           </p>
         </>
       )}
+
       <form
         className="w-full flex flex-col items-center"
-        onSubmit={generateInvoiceHandler}
+        onSubmit={handleSubmit(onSubmit)}
       >
-        {isLoading && (
-          <div className="p-5">
-            <LoadingSpinner />
-          </div>
-        )}
-        {showLnInvoice && (
-          <div className="flex flex-col pb-5 justify-center text-center">
-            <AmountInput amount={amount} onChangeAmount={amountChangeHandler} />
-            <div className="flex flex-col justify-center mt-2">
-              <label
-                htmlFor="comment"
-                className="block text-gray-700 dark:text-white text-sm mb-2 font-bold text-left"
-              >
-                Comment
-              </label>
-              <input
-                id="comment"
-                type="text"
-                placeholder="Optional comment"
-                value={comment}
-                onChange={commentChangeHandler}
-                className="input-underline"
+        <fieldset className="w-4/5 mb-5">
+          {isLoading && (
+            <div className="p-5">
+              <LoadingSpinner />
+            </div>
+          )}
+
+          {showLnInvoice && (
+            <div className="flex flex-col pb-5 justify-center text-center">
+              <AmountInput
+                amount={amount}
+                register={register("amountInput", {
+                  required: t(
+                    "forms.validation.chainAmount.required"
+                  ) as string,
+                  onChange: amountChangeHandler,
+                })}
+                errorMessage={errors.amountInput}
               />
+
+              <div className="flex flex-col justify-center mt-2">
+                <InputField
+                  {...register("commentInput", {
+                    onChange: commentChangeHandler,
+                  })}
+                  label={t("tx.comment")}
+                  value={comment}
+                  placeholder="Optional comment"
```

Comment:
Please do not hardcode english part 2 :)

Reply:
Hatte das aus dem `master` uebernommen :P fixed